File: 2018 KAKAO BLIND RECRUITMENT [1차] 다트 게임.py
def solution(dartResult):
    answer = 0
    result = []
    for i in range(len(dartResult)):
        a = dartResult[i]
        if a == '0' and dartResult[i-1]=='1':
            continue
        if a.isnumeric():
            if dartResult[i+1] == '0':
                result.append(10)
                i += 1
            else:
                result.append(int(a))
                continue
        if a == 'D':
            result[-1] = result[-1]**2
        elif a == 'T':
            result[-1] = result[-1]**3
        elif a == '*':
            if len(result) > 1:
                result[-2] = result[-2]*2
                result[-1] = result[-1]*2
            else:
                result[-1] = result[-1]*2
        elif a == '#':
            result[-1] = result[-1]*(-1)
    answer = sum(result)
            
    return answer

File: 2018 KAKAO BLIND RECRUITMENT [1차] 다트 게임.js
function solution(dartResult) {
  var answer = 0;
  let arr = [];
  for (let i = 0; i < dartResult.length; i++) {
    if ("1234567890".includes(dartResult[i])) {
      if (dartResult[i + 1] === "0") {
        arr.push(10);
        i++;
      } else arr.push(Number(dartResult[i]));
    } else if (dartResult[i] === "D") {
      arr[arr.length - 1] *= arr[arr.length - 1];
    } else if (dartResult[i] === "T") {
      arr[arr.length - 1] *= arr[arr.length - 1] * arr[arr.length - 1];
    } else if (dartResult[i] === "*") {
      if (arr.length > 1) {
        arr[arr.length - 1] *= 2;
        arr[arr.length - 2] *= 2;
      } else {
        arr[arr.length - 1] *= 2;
      }
    } else if (dartResult[i] === "#") {
      arr[arr.length - 1] *= -1;
    }
  }
  for (let i = 0; i < arr.length; i++) {
    answer += arr[i];
  }
  return answer;
}
